VersionControlUI/src/DiffManager.cpp

```diff
@@ -0,0 +1,288 @@
+ **
+ ** Copyright (c) 2011, 2016 ETH Zurich
+ ** All rights reserved.
+ **
+ ** Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+ ** following conditions are met:
+ **
+ **    * Redistributions of source code must retain the above copyright notice, this list of conditions and the
+ **      following disclaimer.
+ **    * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+ **      following disclaimer in the documentation and/or other materials provided with the distribution.
+ **    * Neither the name of the ETH Zurich nor the names of its contributors may be used to endorse or promote products
+ **      derived from this software without specific prior written permission.
+ **
+ **
+ ** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ ** INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ ** DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ ** SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ ** SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ ** WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ ** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **
+ **********************************************************************************************************************/
+
+#include "DiffManager.h"
+#include "VersionControlUIException.h"
+
+#include "VisualizationBase/src/VisualizationManager.h"
+#include "VisualizationBase/src/items/ViewItem.h"
+#include "VisualizationBase/src/nodes/ViewItemNode.h"
+#include "VisualizationBase/src/ViewItemManager.h"
+#include "VisualizationBase/src/declarative/GridLayouter.h"
+#include "VisualizationBase/src/overlays/HighlightOverlay.h"
+
+#include "FilePersistence/src/version_control/GitRepository.h"
+#include "FilePersistence/src/simple/SimpleTextFileStore.h"
+
+#include "OOModel/src/declarations/Project.h"
+#include "OOModel/src/statements/ExpressionStatement.h"
+#include "OOModel/src/expressions/EmptyExpression.h"
+
+#include "ModelBase/src/model/AllTreeManagers.h"
+#include "ModelBase/src/model/TreeManager.h"
+#include "ModelBase/src/nodes/Reference.h"
+
+namespace VersionControlUI
+{
+
+struct ChangeWithNodes {
+		ChangeWithNodes(Model::Node* oldNode, Model::Node* newNode,
+							 FilePersistence::ChangeType changeType)
+			: oldNode_{oldNode}, newNode_{newNode}, changeType_{changeType}
+		{}
+
+		Model::Node* oldNode_;
+		Model::Node* newNode_;
+		FilePersistence::ChangeType changeType_;
```

Comment:
Default initialization means initialize with the default value and you can use `{}` for that. You should always initialize primitive types, pointers and enums. You can use the default initializer, which for enums will pick the first value, but if you want to initialize them to a specific value, then just use it in the initializer: `{MyValue}`.